common/Geometry2d/Arc.cpp

```diff
@@ -10,24 +10,25 @@ namespace Geometry2d {
 
 vector<Point> Arc::intersects(const Line& line) const {
     // http://mathworld.wolfram.com/Circle2d-LineIntersection.html
-    float cx = _center.x();
-    float cy = _center.y();
+    const auto cx = static_cast<float>(_center.x());
+    const auto cy = static_cast<float>(_center.y());
 
-    float x1 = line.pt[0].x() - cx;
-    float y1 = line.pt[0].y() - cy;
-    float x2 = line.pt[1].x() - cx;
-    float y2 = line.pt[1].y() - cy;
+    const auto x1 = static_cast<float>(line.pt[0].x() - cx);
+    const auto y1 = static_cast<float>(line.pt[0].y() - cy);
+    const auto x2 = static_cast<float>(line.pt[1].x() - cx);
+    const auto y2 = static_cast<float>(line.pt[1].y() - cy);
```

Comment:
What's the current C++ thoughts behind using auto? I only use it for iterators or for each loops because I feel like it makes reading code somewhat difficult when types aren't obvious.

Reply:
The warning for here is that the variable is defined with a cast, and so having auto means you don't need to write the type twice. [link](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html#cast-expressions)

Reply:
Whoops, this was more in general, not this specific cast

Reply:
oh. I think the community is still divided between always use auto and like just using it for iterators and stuff.

Reply:
Personally I'm more on the side of only for long/opaque types. Regards it's pretty much a judgment call.